### ISSUE TITLE
Redundant decoder in ESPnetASRModel when ctc_weight is 1.0

### ISSUE THREAD

**Issue by pyf98** [espnet/espnet#4908]
**Describe the bug**
When `ctc_weight=1.0`, the ESPnetASRModel should not have a decoder in its parameters, regardless of the decoder config. However, now it will add a default RNN decoder.

Note that the typical way to use CTC only is to set `ctc_weight: 1.0` and remove `decoder` and `decoder_conf` in the training config.

I believe this bug is introduced in a recent commit, because it worked well previously.

https://github.com/espnet/espnet/blob/master/espnet2/asr/espnet_model.py#L146

It should be something like this:
```
            if ctc_weight == 1.0:
                self.decoder = None
            else:
                self.decoder = decoder
```

**Comment by sw005320** [comment]
Thanks!
@ftshijt, can you fix it?
Also, is there a way to add a test to ensure this does not happen?

**Comment by ftshijt** [comment]
I changed the previous behavior mainly because the `self.decoder = None` is not user-aware from the config (e.g., user can still set decoder with anything, but eventually, there is no decoder is initialized). 

For that purpose, I add a `None` decoder and ask the user to set `decoder=None` when they do the initialization. So probably, we may just throw an error (e.g., you should set decoder to none etc.).

What's your opinion on it?

**Comment by pyf98** [comment]
This will affect our previous CTC configs and models.

**Comment by ftshijt** [comment]
I see. So I will add a warning msg and recover the previous options. thanks for bringing up the issue.

**Comment by pyf98** [comment]
Actually, I think the decoder should be `None` by default. The user should specify a decoder when they really want to use one.

**Comment by ftshijt** [comment]
Yeah, I thought that would be better, but the only concern I had at the time is that, it may lead to some issues to other parts (e.g., test functions etc.) Anyway, since we brought up the issue, let's just do it and fixes issues with the change.
